### PR TITLE
Update keras-cifar-main to user sparse data and loss function.

### PR DESCRIPTION
**Pull request by pavithrasv**
keras-cifar-main synthetic data generates sparse data, whereas the real data generates one-hot labels. The real data and model loss/metric are correct but the synthetic data is wrong with respect to this model parameters. We can convert the synthetic data to return one-hot labels but this synthetic data is shared across other models where sparse data is used. I figured it is best to convert keras-cifar-main to use sparse data as well.

### CODE REVIEW

**Comment by tfboyd** [comment]
I will check it out.  We just want it to match resnet50.

**Comment by tfboyd** [comment]
I will kick off a quick benchmark as well.

**Comment by pavithrasv** [comment]
Thank you Toby!

File: official/resnet/keras/keras_cifar_main.py [hunk 1]
```diff
@@ -82,9 +82,7 @@ def parse_record_keras(raw_record, is_training, dtype):
   Returns:
     Tuple with processed image tensor and one-hot-encoded label tensor.
   """
-  image, label = cifar_main.parse_record(raw_record, is_training, dtype)
-  label = tf.compat.v1.sparse_to_dense(label, (cifar_main.NUM_CLASSES,), 1)
-  return image, label
+  return cifar_main.parse_record(raw_record, is_training, dtype)
```
Comment (tfboyd) [inline official/resnet/keras/keras_cifar_main.py]: While I do a test, can you remove this method all together so it is not a pass through.

Comment (pavithrasv) [reply official/resnet/keras/keras_cifar_main.py]: Done, thank you!

**Comment by tfboyd** [comment]
Taking me longer than I would like.  Haoyu also said this looks ok.  I needed to finish the LSTM example first.  It may take until next week to validate performance and double check convergence.

**Comment by pavithrasv** [comment]
Thank you for the update!

**Comment by pavithrasv** [comment]
Hi all, just wanted to check if there is an update on this and if we can merge this change, thank you!

**Comment by pavithrasv** [comment]
did the convergence tests pass? can we merge this change?

**Comment by guptapriya** [comment]
@saberkun can someone from the model garden help review and validate this change?

**Comment by pavithrasv** [comment]
This is blocking a bug fix change, would like to submit that by end of this week if possible. Thank you!

**Comment by saberkun** [comment]
Hi, pavithrasv@
Could you please resolve the conflict first?
Please ping me internally. Note that, you are free to work on internal repo without hassle to sync to github.

**Comment by pavithrasv** [comment]
Submitted this change internally.
